### ISSUE TITLE
After applying provisioning template document templates don't work (opens blank window)

### ISSUE THREAD

**Issue by kucerasolitea** [SharePoint/PnP-Sites-Core#2054]
#### Which PnP repository should you use to report the issue? 
- PnP Sites Core - https://github.com/OfficeDev/PnP-Sites-Core

#### Category
[x] Bug
[ ] Enhancement

#### Environment
[x] Office 365 / SharePoint Online
[ ] SharePoint 2016
[ ] SharePoint 2013

#### Expected or Desired Behavior
New document templates work as expected after applying provisioning template on a new site

#### Observed Behavior
Document templates are not working after applying provisioning template on a new site
After clicking a New button -> Document (template) it opens just blank window

#### Steps to Reproduce
1.Create a new site collection (Modern team site)
2.Create a new document library 'Test'
3.Go to the library and add document template to the library using New button -> Add a template like 'Document.docx'
4.Create a provisiong template using cmdlet [Get-PnPProvisioningTemplate -out template.xml]
5.Create a new site collection (Modern team site)
6.Apply the provisioning template on newly created site
7.Go to the document library and click New button -> 'Document' to create a new Word document based on the template

If I follow these steps I am not able to get document templates to work.
As a heads up, we use localized tenant (Czech).

Looking into provisioning template xml file, I could see links referencing document templates located in the original site, see snippet of code from my xml file:

<NewDocumentTemplates>[{"templateId":"NewFolder","title":"Složka","visible":true},{"contentTypeId":"0x0101000B31B17C8DD73D488B05B5E1FE6FF3A900EBD6F4AECECBE94B83DA3F6E3E340CE200637D8EA722B66240A6DE1062153B73EA","isContentType":true,"templateId":"0x0101000B31B17C8DD73D488B05B5E1FE6FF3A900EBD6F4AECECBE94B83DA3F6E3E340CE200637D8EA722B66240A6DE1062153B73EA","title":"Cílový koncept","visible":false},{"templateId":"NewDOC","title":"Wordový dokument","visible":false},{"templateId":"NewXSL","title":"Excelový sešit","visible":false},{"templateId":"NewPPT","title":"Powerpointová prezentace","visible":false},{"templateId":"NewONE","title":"Onenotový poznámkový blok","visible":false},{"templateId":"NewXSLForm","title":"Forms pro Excel","visible":false},{"templateId":"Link","title":"Odkaz","visible":false},{"contentTypeId":"0x0101000B31B17C8DD73D488B05B5E1FE6FF3A90051E71A5FF63C8A44B5710EE26EE7D8B8003F830C8E0D5B3A4498FD2D1644101A34","isUpload":true,"templateId":"Zápis z workshopu","title":"Zápis z workshopu","url":"/sites/CDL-DI-ProjektovyWebSablona/BluePrint/Forms/Zápis z workshopu.docx","visible":false},{"contentTypeId":"0x0101000B31B17C8DD73D488B05B5E1FE6FF3A90051E71A5FF63C8A44B5710EE26EE7D8B8003F830C8E0D5B3A4498FD2D1644101A34","isUpload":true,"templateId":"Harmonogram_WS","title":"Harmonogram_WS","url":"/sites/CDL-DI-ProjektovyWebSablona/BluePrint/Forms/Harmonogram_WS.xlsx","visible":false},{"contentTypeId":"0x0101000B31B17C8DD73D488B05B5E1FE6FF3A9002FFE389177AA594AA9543459762CB599002DC72933312A9C44A42B2BAFF7E656EB","isUpload":true,"templateId":"Zápis z workshopu","title":"Zápis z workshopu","url":"/sites/CDL-DI-ProjektovyWebSablona/BluePrint/Forms/Zápis z workshopu.docx","visible":true},{"contentTypeId":"0x0101000B31B17C8DD73D488B05B5E1FE6FF3A9002FFE389177AA594AA9543459762CB599002DC72933312A9C44A42B2BAFF7E656EB","isUpload":true,"templateId":"Harmonogram_WS","title":"Harmonogram_WS","url":"/sites/CDL-DI-ProjektovyWebSablona/BluePrint/Forms/Harmonogram_WS.xlsx","visible":true}]</NewDocumentTemplates>
![template_opens_blank_window](https://user-images.githubusercontent.com/43810855/52052746-aa3cbb00-2557-11e9-86da-b2d1f18d5b7e.png)

**Comment by dharm-4** [comment]
Yes. We are also facing same issue with custom template provisioning. is there any way to deploy custom created template with site template to new sharepoint site? 
As per our understanding it is maintaining the reference in template XML file but during the provisioning it is not extracting that file for new sharepoint site. So, on the new SPSite it will not find the custom template file and it will only create the reference of that template on UI.

Looking forward for the solution of this problem.

Thanks!

**Comment by meerapan** [comment]
Hey, 

Wonder where you see the code snippet in the template xml. Which node is it in? 
Well, I added  custom templates to a library and extracted the template using cmdlet [Get-PnPProvisioningTemplate -out template.xml] but no markup about the custom templates I added. 

How can we associate a template to a library in the 
PnP provisioning template? Thinking of provisioning the template files to a library in the site and linking them.
